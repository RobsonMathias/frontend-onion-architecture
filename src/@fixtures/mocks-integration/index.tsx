export * from './store.mock';
export * from './adapters.mock';
export * from './../mocks/http.mock';
export * from './../mocks/router.mock';
