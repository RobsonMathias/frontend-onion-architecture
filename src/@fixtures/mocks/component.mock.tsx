import React from 'react';

export function Mock() {
  return <div />;
}

export const mockComponent = (componentsName: string[] = []) => {
  let composed: { [key: string]: React.ReactNode } = {};
  componentsName.forEach((name: string) => {
    composed[name] = Mock;
  });
  return composed;
};
