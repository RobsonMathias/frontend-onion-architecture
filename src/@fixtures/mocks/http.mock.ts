export const mockHttp = {
  patch: jest.fn(),
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
  delete: jest.fn(),
  interceptors: {
    request: {
      use: jest.fn(),
      eject: jest.fn()
    },
    response: {
      use: jest.fn()
    }
  }
} as any;
