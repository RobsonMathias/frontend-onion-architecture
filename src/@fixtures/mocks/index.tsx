export * from './store.mock';
export * from './router.mock';
export * from './component.mock';
export * from './wrappers';
export * from './http.mock';
export * from './services.mock';
export * from './navigation.mock';
export * from './storage.mock';
export * from './use-auth.mock';
