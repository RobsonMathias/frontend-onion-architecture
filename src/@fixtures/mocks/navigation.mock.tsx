import React from 'react';

const setData = jest.fn();
const mockUseUI = jest.fn(() => ({ setData }));

const query = { get: jest.fn(), set: jest.fn() };
const set = jest.fn();
const setAll = jest.fn();
const mockUseQuery = jest.fn(() => ({ query, setAll, set }));

const replace = jest.fn();
const mockUseURL = jest.fn(() => ({ replace }));

const mockNavBar = React.forwardRef((props, ref) => <div {...props} />);

export const mockNavigation = {
  NavBar: mockNavBar,
  useQuery: mockUseQuery,
  query,
  setAll,
  set,
  useUI: mockUseUI,
  setData,
  useURL: mockUseURL,
  replace
};
