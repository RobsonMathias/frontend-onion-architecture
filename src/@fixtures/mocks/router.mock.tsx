import React from 'react';

function Mock({ children }: any) {
  return <>{children}</>;
}

const mockUseLocation = jest.fn(() => ({}));
const goBack = jest.fn();
const push = jest.fn();
const listen = jest.fn();
const mockUseHistory = jest.fn(() => ({ push, goBack, location: {}, listen, action: 'ACTION' }));
const useRouteMatch = () => ({ push: jest.fn(), url: '/mocked' });
const mockUseParams = jest.fn();
const mockNavLink = React.forwardRef(
  ({ to, children, activeClassName, className, 'data-testid': testId }: any, ref) => (
    <a
      className={`${className}-${activeClassName}`}
      href={to}
      children={children}
      data-testid={testId}
    />
  )
);
const Link = React.forwardRef((props, ref) => <p {...props} />);
const mockRoute = React.forwardRef((props, ref) => <div />);
const mockSwitch = React.forwardRef((props, ref) => <div />);
jest.mock('react-router-dom', () => ({
  Redirect: Mock,
  BrowserRouter: Mock,
  useHistory: mockUseHistory,
  useParams: mockUseParams,
  useRouteMatch,
  NavLink: mockNavLink,
  Route: mockRoute,
  Switch: mockSwitch,
  useLocation: mockUseLocation
}));

export const mockRouter = {
  useHistory: mockUseHistory,
  Link,
  useParams: mockUseParams,
  NavLink: mockNavLink,
  useLocation: mockUseLocation,
  goBack,
  push,
  listen
};
