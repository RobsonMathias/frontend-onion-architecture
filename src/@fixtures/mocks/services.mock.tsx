import { Service } from 'services/service';

const auth = {
  onAuthStateChange: jest.fn(),
  signOut: jest.fn(),
  signIn: jest.fn(),
  signInWithGoogle: jest.fn(),
  signUp: jest.fn()
};

const task = {
  list: jest.fn(),
  create: jest.fn(),
  update: jest.fn(),
  delete: jest.fn()
};

function ServicesProvider({ children }: any) {
  return <div>{children}</div>;
}

const instances = {
  auth,
  task
};
const mock = {
  ServicesProvider,
  useService: jest.fn(() => instances)
};
jest.doMock('services', () => mock);

export const mockServices = {
  mock,
  auth,
  task
};

export const mockServiceInstance = {
  patch: jest.fn(),
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
  delete: jest.fn(),
  setInterceptorHeaders: jest.fn(),
  removeInterceptorHeaders: jest.fn()
} as any;
