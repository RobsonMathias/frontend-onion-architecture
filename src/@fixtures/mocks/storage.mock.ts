export const storage = {
  get: jest.fn(),
  save: jest.fn(),
  delete: jest.fn()
} as any;

export const localStorage = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  clear: jest.fn(),
  key: jest.fn(),
  removeItem: jest.fn(),
  length: jest.fn()
} as any;
