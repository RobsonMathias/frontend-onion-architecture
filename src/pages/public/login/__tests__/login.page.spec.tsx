/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';

import { mockUseAuth, Mock } from '@fixtures/mocks';
import { render } from '@testing-library/react';
import React from 'react';

jest.mock('modules/auth', () => ({
  useAuth: jest.fn().mockImplementation(() => mockUseAuth()),
  LoginContainer: Mock,
  GoogleContainer: Mock
}));

import LoginPage from '../login.page';

describe('LoginPage', () => {
  it('should render successfully', () => {
    const { container } = render(<LoginPage />);
    expect(container).toMatchSnapshot();
  });
});
