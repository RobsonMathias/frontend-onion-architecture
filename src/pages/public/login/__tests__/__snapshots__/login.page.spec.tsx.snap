// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render successfully 1`] = `
<div>
  <div
    class="MuiBox-root css-36s324"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-10ib5jr"
        >
          <h1
            class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
          >
            Ol√°
          </h1>
        </div>
        <div />
        <div
          class="MuiBox-root css-131uyni"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-overline MuiTypography-alignCenter css-e1onu5-MuiTypography-root"
              >
                OU
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              />
            </div>
          </div>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
